File: ios/Domain/Models/BarcodeScanningResult.swift
import Foundation

#if MLKIT_BARCODE_SCANNING
  struct BarcodeScanningResult {
    let barcodes: [BarcodeData]
  }

  struct BarcodeData {
    let bounds: BoundingBox?
    let corners: [Corner]?
    let format: Int
    let formatName: String
    let valueType: Int
    let valueTypeName: String
    let rawValue: String?
    let displayValue: String?
    let rawBytes: [Int]?
    let isPotential: Bool
    let value: BarcodeParsedValue?
  }

  struct BarcodeParsedValue {
    let type: String
    let data: [String: Any?]
  }
#endif  // MLKIT_BARCODE_SCANNING

File: android/src/main/java/com/visioncameramlkit/domain/models/BarcodeScanningResult.kt
package com.visioncameramlkit.domain.models

data class BarcodeScanningResult(
  val barcodes: List<BarcodeData>,
)

data class BarcodeData(
  val bounds: BoundingBox?,
  val corners: List<Corner>?,
  val format: Int,
  val formatName: String,
  val valueType: Int,
  val valueTypeName: String,
  val rawValue: String?,
  val displayValue: String?,
  val rawBytes: List<Int>?,
  val isPotential: Boolean,
  val value: BarcodeParsedValue?,
)

data class BarcodeParsedValue(
  val type: String,
  val data: Map<String, Any?>,
)
